ignite/handlers/checkpoint.py

```diff
@@ -178,3 +181,100 @@ def __call__(self, engine, to_save):
             _, paths = self._saved.pop(0)
             for p in paths:
                 os.remove(p)
+
+
+class EngineCheckpoint(object):
+    """EngineCheckpoint handler can be used to periodically save engine and other objects
+     to disk. Saved checkpoint then can be used to resume Engine.
+
+    Args:
+        dirname (str):
+
+
+    """
+
+    def __init__(self, dirname, to_save,
+                 save_interval=100,
+                 atomic=True, create_dir=True):
+        assert isinstance(dirname, str)
+        assert isinstance(to_save, dict), "Argument `to_save` should be a dictionary"
+        self.check_objects(to_save)
+
+        self.dirname = dirname
+        self.to_save = to_save
+        self.save_interval = save_interval
```

Comment:
Why do we have this attribute twice?

Reply:
@alykhantejani you are right my bad, copy paste